src/main/java/org/truffleruby/extra/ffi/Pointer.java

```diff
@@ -135,12 +135,7 @@ public double readDouble(long offset) {
     }
 
     public Pointer readPointer(long offset) {
-        final long p = readLong(offset);
-        if (p == 0) {
-            return null;
-        } else {
-            return new Pointer(p);
-        }
+        return new Pointer(readLong(offset));
```

Comment:
Do we care about reference equality? If so, you probably want to keep the branch but return `Pointer.NULL`.

Reply:
No, I think it was code to be compatible with JNR, now it doesn't matter and the identity of Pointer is never used afaik.